extensions/metadata/interface/src/main/java/org/exist/storage/md/CollectionEvents.java

```diff
@@ -29,113 +29,142 @@
-import org.exist.storage.lock.Lock;
 import org.exist.storage.txn.Txn;
 import org.exist.xmldb.XmldbURI;
 
 /**
  * @author <a href="mailto:shabanovd@gmail.com">Dmitriy Shabanov</a>
- *
+ * 
  */
 public class CollectionEvents implements CollectionTrigger {
 
-	@Override
-	public void configure(DBBroker broker, Collection parent, Map<String, List<? extends Object>> parameters) throws TriggerException {
-	}
-
-	@Override
-	public void beforeCreateCollection(DBBroker broker, Txn txn, XmldbURI uri) throws TriggerException {
-//		System.out.println("beforeCreateCollection "+uri);
-	}
-
-	@Override
-	public void afterCreateCollection(DBBroker broker, Txn txn, Collection collection) throws TriggerException {
-//		System.out.println("afterCreateCollection "+collection.getURI());
-		try {
-			MDStorageManager.inst.md.addMetas(collection);
-		} catch (Throwable e) {
-			MDStorageManager.LOG.fatal(e);
-		}
-	}
-
-	@Override
-	public void beforeCopyCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI newUri) throws TriggerException {
-	}
-
-	@Override
-	public void afterCopyCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI oldUri) throws TriggerException {
-//		System.out.println("afterCopyCollection "+collection.getURI());
-		try {
-			MDStorageManager.inst.md.copyMetas(oldUri, collection);
-		} catch (Throwable e) {
-			MDStorageManager.LOG.fatal(e);
-		}
-	}
-
-	@Override
-	public void beforeMoveCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI newUri) throws TriggerException {
-//		System.out.println("beforeMoveCollection "+collection.getURI());
-		try {
-	        for(Iterator<DocumentImpl> i = collection.iterator(broker); i.hasNext(); ) {
-	            DocumentImpl doc = i.next();
-	            MDStorageManager.inst.md.moveMetas(
-            		collection.getURI().append(doc.getFileURI()), 
-            		newUri.append(doc.getFileURI())
-        		);
-	        }
-		} catch (PermissionDeniedException e) {
-			throw new TriggerException(e);
-		}
-	}
-
-	@Override
-	public void afterMoveCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI oldUri) throws TriggerException {
-//		System.out.println("afterMoveCollection "+oldUri+" to "+collection.getURI());
-		MDStorageManager.inst.md.moveMetas(oldUri, collection.getURI());
-	}
-	
-	private void deleteCollectionRecursive(DBBroker broker, Collection collection) throws PermissionDeniedException {
-        for(Iterator<DocumentImpl> i = collection.iterator(broker); i.hasNext(); ) {
-            DocumentImpl doc = i.next();
-            MDStorageManager.inst.md.delMetas(doc.getURI());
+    @Override
+    public void configure(DBBroker broker, Collection parent, Map<String, List<? extends Object>> parameters) throws TriggerException {
+    }
+
+    @Override
+    public void beforeCreateCollection(DBBroker broker, Txn txn, XmldbURI uri) throws TriggerException {
+        // System.out.println("beforeCreateCollection "+uri);
```

Comment:
@adamretter I use it for debugging, add it each time I start debugging slow down a lot